rust/src/parse.rs

```diff
@@ -0,0 +1,392 @@
+//! Functions for parsing byte buffers into their MCAP record contents.
+use crate::records::*;
+use std::borrow::Cow;
+use std::collections::BTreeMap;
```

Comment:
Why BTreeMap?

<img width="675" alt="image" src="https://user-images.githubusercontent.com/14237/189775531-3a7780fa-0ab7-4c9a-abfa-9f91d98d2465.png">

Reply:
Not a strong preference. I want the order of keys to be stable and well-defined, though I can't really put into words why - seems good for test determinism.